std/algorithm/iteration.d

```diff
@@ -1502,9 +1502,12 @@ if (isInputRange!R)
     static if (isForwardRange!R)
     {
         ///
-        @property typeof(this) save() {
+        @property typeof(this) save()
+        {
+            import std.algorithm.mutation : move;
             typeof(this) ret = this;
-            ret._input = this._input.save;
+            auto saved = this._input.save;
+            move(saved, ret._input);
             ret._current = this._current;
```

Comment:
shouldn't this line also invoke refRange's opAssign?

Reply:
> shouldn't this line also invoke refRange's opAssign?

`_current` is the `front` element. It's not the range itself.